Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/SqlDialect.cs

```diff
@@ -50,5 +53,65 @@ public virtual string DelimitIdentifier (string identifier)
       return "[" + identifier + "]";
     }
 
+    public IDbDataParameter CreateDataParameter (IDbCommand command, IStorageTypeInformation storageTypeInformation, string parameterName, object? value)
+    {
+      ArgumentUtility.CheckNotNull("command", command);
+      ArgumentUtility.CheckNotNull("storageTypeInformation", storageTypeInformation);
+      ArgumentUtility.CheckNotNullOrEmpty("parameterName", parameterName);
+
+      var convertedValue = storageTypeInformation.ConvertToStorageType(value);
+
+      var parameter = command.CreateParameter();
+      parameter.ParameterName = GetParameterName(parameterName);
+      parameter.Value = convertedValue;
+      parameter.DbType = storageTypeInformation.StorageDbType;
+      if (storageTypeInformation.StorageTypeLength.HasValue)
+      {
+        var parameterSize = storageTypeInformation.StorageTypeLength.Value;
+        if (parameterSize == SqlStorageTypeInformationProvider.StorageTypeLengthRepresentingMax)
+        {
+          var fulltextCompatibleMaxLength = storageTypeInformation.StorageDbType switch
+          {
+              DbType.AnsiString => 8000,
+              DbType.String => 4000,
+              // AnsiStringFixedLength and StringFixedLength represent char(...) and nchar(...), which do not support "max" as size
```

Comment:
We also had explciit tests for the FixedLength types to better document the intended behavior. I believe those are gone now?

Reply:
Those did not exist